tests/unit-tests/src/VuFindTest/OaiPmh/HarvesterTest.php

```diff
@@ -338,6 +338,45 @@ public function testListRecordsWithResumption()
         $harvester->launch();
     }
 
+    /**
+     * Test that we only retrieve one page of results
+     * if ignoreResumtionToken is set.
+     *
+     * @return void
+     */
+    public function testListRecordsWithIgnoreResumptionTokenOption()
+    {
+        $comm = $this->getMockCommunicator();
+        $expectedSettings0 = [
+            'metadataPrefix' => 'oai_dc', 'set' => 'xyzzy',
+            'from' => '2016-07-01', 'until' => '2016-07-31',
+        ];
+        $expectedSettings1 = ['resumptionToken' => 'more'];
+        $comm->expects($this->exactly(1))->method('request')
+            ->withConsecutive(
+                ['ListRecords', $expectedSettings0],
+            )
+            ->willReturnOnConsecutiveCalls(
+                simplexml_load_string($this->getFakeResponse())
+            );
+        $writer = $this->getMockRecordWriter();
+        $writer->expects($this->exactly(1))->method('write')
+            ->with($this->isInstanceOf('SimpleXMLElement'));
+        $sm = $this->getMockStateManager();
+        $sm->expects($this->once())->method('clearState');
```

Comment:
You might want to also add here:

```
$sm->expects($this->never())->method('saveState');
```

This will just confirm that the state save step does not happen here, since we expect that it shouldn't.